spaceapi_server/src/lib.rs

```diff
@@ -140,23 +138,16 @@ fn status_endpoint(_: Request, mut res: Response<Fresh>) {
 
     // Get response body
     let body = build_response_json(people_present, raspi_temperature);
-    let body_bytes = body.as_bytes();
 
-    // Set headers
-    // A new scope is used here because of the mutable borrow.
-    {
-        let mut headers = res.headers_mut();
-        headers.set(header::ContentLength(body_bytes.len() as u64));
-        headers.set(header::ContentType("application/json; charset=utf-8".parse().unwrap()));
-        headers.set(header::CacheControl(vec![header::CacheDirective::NoCache]));
-        headers.set(header::AccessControlAllowOrigin::Any);
-    }
+    // Create response
+    let mut response = Response::with((status::Ok, body));
 
-    // Write response body
-    let mut stream = res.start().unwrap();
-    stream.write_all(body_bytes).unwrap();
+    // Set headers
+    response.set_mut(Header(headers::ContentType("application/json; charset=utf-8".parse().unwrap())));
+    response.set_mut(Header(headers::CacheControl(vec![headers::CacheDirective::NoCache])));
+    response.set_mut(Header(headers::AccessControlAllowOrigin::Any));
```

Comment:
Here, instead of using `set_mut` we could also use chained `set` calls on the response, in which case it wouldn't need to be mutable.

But this way it's a bit nicer to structure the code.

If you think the chained approach would be better, let me know.